packages/salesforcedx-vscode-core/src/commands/forceAuthDevHub.ts

```diff
@@ -37,6 +48,51 @@ export class ForceAuthDevHubExecutor extends SfdxCommandletExecutor<{}> {
     command.withArg('--setdefaultdevhubusername');
     return command.build();
   }
+
+  public async execute(response: ContinueResponse<any>): Promise<void> {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: getRootWorkspacePath()
+    }).execute(cancellationToken);
+
+    execution.processExitSubject.subscribe(this.configureDefaultDevHubLocation.bind(this));
+
+    this.attachExecution(execution, cancellationTokenSource, cancellationToken);
+  }
+
+  public async configureDefaultDevHubLocation() {
+    const globalDevHubName = await OrgAuthInfo.getDefaultDevHubUsernameOrAlias(
+      false,
+      ConfigSource.Global
+    );
+
+    if (isNullOrUndefined(globalDevHubName)) {
+      const localDevHubName = await OrgAuthInfo.getDefaultDevHubUsernameOrAlias(
+        false,
+        ConfigSource.Local
+      );
+
+      if (localDevHubName) {
+        await this.setGlobalDefaultDevHub(localDevHubName);
+      }
+    }
+  }
+
+  public async setGlobalDefaultDevHub(newUsername: string): Promise<void> {
+
+    const homeDirectory = require('os').homedir();
```

Comment:
If we need to import 'os' let's include it with the rest of the imports at the top of the file

Reply:
^ and either `import { homedir } from 'os';` or `import * as os from 'os'`